src/index.js

```diff
@@ -1,16 +1,85 @@
-const apiCall = () => {
-	fetch(`https://api.openweathermap.org/data/2.5/weather?q=London,uk&APPID=${process.env.MY_API_KEY}`, {mode: 'cors'})
-    .then(function(response) {
-      return response.json()
-    })
-    .then(function(response) {
-      const weather = {
-      	temp: response.main.temp,
-      	description: response.weather[0].description,
-      	icon: response.weather[0].icon
-      }
-      console.log(weather)
-      return weather
-    })
+import './scss/index.scss';
+
+const apiCall = async (city, unit) => {
+	try {
+		const resp = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${city}&units=${unit}&APPID=${process.env.MY_API_KEY}`, {mode: 'cors'})
+		const data = await resp.json()
+		const weather = {
+			city: data.name,
+	      	temp: Math.round(data.main.temp),
+	      	description: data.weather[0].description,
+	      	wind: data.wind.speed,
+	      	iconId: data.weather[0].icon
+	    }
+		displayWeather(weather)
+	} catch(err) {
+		handleError()
+	}
 }
-apiCall()
\ No newline at end of file
+
+
+const metric = {
+	system: "metric",
+	icon: "C",
+	name: 'Celcius'
+}
+
+const imperial = {
+	system: "imperial",
+	icon: "F",
+	name: 'Fahrenheit'
+}
+
+let currentUnit = metric
+
+const handleError = () => {
+	const city = document.getElementById('cityName')
+	const temp = document.getElementById('temp')
+	const description = document.getElementById('desc')
+	const wind = document.getElementById("wind")
+	city.textContent = 'City not found'
+	img.src = ''
+	temp.textContent = null
+	description.textContent = null
+	wind.textContent = null
+}
+
+const getWeather = () => {
+	const city = document.getElementById('city').value
+	apiCall(city, currentUnit.system )
+}
+
+const displayWeather = (weather) => {
+	const weatherData = document.getElementById('weatherData')
+	const temp = document.getElementById('temp')
+	const city = document.getElementById('cityName')
```

Comment:
Where this `city` variable used?

Reply:
As far as I understand I was using element's id "cityName" instead of city variable later by accident. I worked though, never knew that was an option. Is that a good practice though?

Reply:
Ask yourself a simple question? "How much easier it is for me if I want to modify something in my code?" In an example suppose I need `city` element in multiple places. Now if I need to write `document.getElementById(cityName')` multiple time that means it's doing that same operation multiple time which may affect my apps performances. Also, suddenly, I changed my mind. I changed that ID with something else, now it's quite hard for me to figure out where I used that `document.getElementById(cityName')` so that I can update. Now, If I put this thing in a variable no matter how many times I used it. It will help me to organize my code and maintain my code.